rinex-cli/src/graph/record/navigation.rs

```diff
@@ -99,119 +192,200 @@ pub fn plot_sv_nav_clock(ctx: &RnxContext, plot_ctx: &mut PlotContext) {
+                    );
+                },
+                ProductType::HighPrecisionClk => {
+                    plot_ctx.add_timedomain_plot(
+                        &format!("|{} - SV| CLK Correction", ts),
+                        "Offset [s]",
+                    );
+                },
+            }
+            for (index, (sv, data)) in vehicles
+                .iter()
+                .filter(|(sv, data)| sv.constellation.timescale().unwrap() == ts)
+                .enumerate()
+            {
+                let epochs = data.iter().map(|(t, _)| *t).collect::<Vec<_>>();
+                let offset = data
+                    .iter()
+                    .map(|(_, dt)| dt.to_seconds())
+                    .collect::<Vec<_>>();
                 let trace =
-                    build_chart_epoch_axis(&format!("{}", sv), Mode::Markers, epochs, clock_corr)
+                    build_chart_epoch_axis(&format!("{}", sv), Mode::Markers, epochs, offset)
                         .visible({
-                            if sv_index < 3 {
+                            if index == 0 {
                                 Visible::True
                             } else {
                                 Visible::LegendOnly
                             }
                         });
                 plot_ctx.add_trace(trace);
             }
-        } else {
-            warn!("cannot plot brdc clock correction: needs OBS RINEX");
+        }
+    }
+}
+
+/*
+ * Plot System time as resolved from all provided products
+ */
+fn plot_system_time(
+    states: &CtxClockStates,
+    corrections: &CtxClockCorrections,
+    nav_sv: &Vec<SV>,
+    plot_ctx: &mut PlotContext,
+) {
+    trace!("time system plot");
+    for (product, vehicles) in corrections {
+        for ts in vehicles
+            .iter()
+            .map(|(sv, _)| sv.constellation.timescale().unwrap())
+            .unique()
+        {
+            match product {
+                ProductType::Radio => {
+                    plot_ctx.add_timedomain_plot(&format!("{} from BRDC", ts), "Epoch");
+                },
+                ProductType::HighPrecisionSp3 => {
+                    plot_ctx.add_timedomain_plot(&format!("{} from SP3", ts), "Epoch");
+                },
+                ProductType::HighPrecisionClk => {
+                    plot_ctx.add_timedomain_plot(&format!("{} from CLK", ts), "Epoch");
+                },
+            }
+            for (_, state_vehicles) in states.iter().filter(|(k, _)| *k == product) {
+                for (index, (sv, corrections)) in vehicles
+                    .iter()
+                    .filter(|(sv, data)| sv.constellation.timescale().unwrap() == ts)
+                    .enumerate()
+                {
+                    if let Some((_, data)) = state_vehicles
+                        .iter()
+                        .filter(|(state_sv, data)| *state_sv == sv)
+                        .reduce(|k, _| k)
+                    {}
```

Comment:
This does nothing, maybe add a FIXME?

Reply:
Correct, I'll mark a FIXME, my plan is to conclude these operations/graphs in a separate PR today, because this PR is already slightly off topic and contains good improvements